rust/src/index/vector/graph/builder.rs

```diff
@@ -55,30 +80,54 @@ impl<V: Vertex> GraphBuilder<V> {
         &mut self.nodes[id].vertex
     }
 
-    pub fn neighbors(&self, id: usize) -> &[u32] {
-        self.nodes[id].neighbors.as_slice()
-    }
-
     pub fn neighbors_mut(&mut self, id: usize) -> &mut Vec<u32> {
         &mut self.nodes[id].neighbors
     }
 
-    pub fn add_edge(&mut self, from: usize, to: usize) {
-        self.nodes[from].neighbors.push(to as u32);
+    /// Set neighbors of a node.
+    pub fn set_neighbors(&mut self, id: usize, neighbors: impl Into<Vec<u32>>) {
```

Comment:
should this be vertex as well?

Reply:
vertex? or Graph?

The graph interface will be shared with `persisted graph` as well, so the `greedy_search` will shared between index and graph builder.